src/client.ts

```diff
@@ -105,11 +105,34 @@ export class SkynetClient {
    * The Skynet Client which can be used to access Skynet.
    *
    * @class
-   * @param [portalUrl] The portal URL to use to access Skynet, if specified. To use the default portal while passing custom options, use ""
+   * @param [initialPortalUrl] The initial portal URL to use to access Skynet, if specified. A request will be made to this URL to get the actual portal URL. To use the default portal while passing custom options, pass "".
    * @param [customOptions] Configuration for the client.
    */
-  constructor(portalUrl: string = defaultPortalUrl(), customOptions: CustomClientOptions = {}) {
-    this.portalUrl = portalUrl;
+  constructor(initialPortalUrl = "", customOptions: CustomClientOptions = {}) {
+    // Get the initial portal URL, if not provided.
```

Comment:
what scenario prevents us from defaulting to `defaultPortalUrl` here in the params?